docs/examples/check-email-for-intent.md

```diff
@@ -0,0 +1,243 @@
+    function: imap
+    input:
+      email: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      imap-address: "imap.gmail.com:993"
+  transition: sentiment-check
+```
+
+### Check Intent
+
+```yaml
+- id: sentiment-check
+  type: action
+  action:
+    function: sentiment
+    secrets: ["SERVICE_ACCOUNT_KEY"]
+    input:
+      message: "{{.return.msg}}"
+      serviceAccountKey: "{{.secrets.SERVICE_ACCOUNT_KEY}}"
+  transition: check-feeling
+```
+
+### Check Feeling
+If the feeling of the text written in the email is negative send an automatic response back to the sender.
+
+```yaml
+- id: check-feeling
+  type: switch
+  conditions:
+  - condition: '.return.feeling == "Negative"'
+    transition: send-response
+```
+
+### Reply to Email
+This uses the smtp server to send a response to the person that sent you the email in the first place.
+
+```yaml
+- id: send-response
+  type: action
+  action:
+    secrets: ["EMAIL_NAME","EMAIL_PASSWORD"]
+    function: smtp
+    input:
+      to: "{{.secrets.EMAIL_NAME}}"
+      subject: "I dont appreciate your message"
+      message: ""
+      from: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      server: smtp.gmail.com
+      port: 587
+```
+
+### Workflow Read Email
+
+```yaml
+id: listen-for-email
+description: This workflow reads an email when a cloud event is received.
+functions:
+- id: imap
+  image: vorteil/imap:v1
+- id: smtp
+  image: vorteil/smtp:v3
+- id: sentiment
+  image: vorteil/google-sentiment-check:v2
+start:
+  type: event
+  state: read-email
+  event:
+    type: smtp
+states:
+- id: read-mail
+  type: action
+  action:
+    secrets: ["EMAIL_NAME", "EMAIL_PASSWORD"]
+    function: imap
+    input:
+      email: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      imap-address: "imap.gmail.com:993"
+  transition: sentiment-check
+- id: sentiment-check
+  type: action
+  action:
+    function: sentiment
+    secrets: ["SERVICE_ACCOUNT_KEY"]
+    input:
+      message: "{{.return.msg}}"
+      serviceAccountKey: "{{.secrets.SERVICE_ACCOUNT_KEY}}"
+  transition: check-feeling
+- id: check-feeling
+  type: switch
+  conditions:
+  - condition: '.return.feeling == "Negative"'
+    transition: send-response
+- id: send-response
+  type: action
+  action:
+    secrets: ["EMAIL_NAME", "EMAIL_PASSWORD"]
+    function: smtp
+    input:
+      to: "{{.secrets.EMAIL_NAME}}"
+      subject: "I dont appreciate your message"
+      message: ""
+      from: "{{.secrets.EMAIL_NAME}}"
+      password: "{{.secrets.EMAIL_PASSWORD}}"
+      server: smtp.gmail.com
+      port: 587
+```
+
+This seems like very plain example. But we could extend it to run a 'imap' listener that sends a cloud event instantly to trigger the sentiment checker on Direktiv. Then based on the sentiment check we could either delete the message move to a certain inbox or respond directly back to the sender.
```

Comment:
Although this particular example is quite simple, the logic used could be modified for something far more 'in-depth'. For example, it could be extended to run an 'IMAP' listener that generates a cloud event, triggering the 'intent' checker on Direktiv. Depending on the determined intent of each email received, actions such as email deletion could be performed.